aws/resource_aws_apigatewayv2_authorizer.go

```diff
@@ -32,23 +32,33 @@ func resourceAwsApiGatewayV2Authorizer() *schema.Resource {
 				Optional:     true,
 				ValidateFunc: validateArn,
 			},
+			"authorizer_payload_format_version": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"1.0", "2.0"}, false),
+			},
+			"authorizer_result_ttl_in_seconds": {
```

Comment:
The default as displayed in the console is `300`:

![Screenshot 2020-09-22 090041](https://user-images.githubusercontent.com/2404182/93885485-2eeec680-fcb2-11ea-89e6-c20127b94da3.png)

and and explicit value of `0` disables caching, so I'll set the default in the schema to `300` (and update the documentation) and add an explicit test of setting to `0`.
Thanks.

Reply:
OK, made those changes which involved querying the API protocol type during authorizer creation otherwise we get errors like

```
Error: error creating API Gateway v2 authorizer: BadRequestException: AuthorizerResultTtlInSeconds cannot be set for WEBSOCKET protocol Apis.
```

Re-ran acceptance tests:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSAPIGatewayV2Authorizer_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Authorizer_ -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Authorizer_basic
=== PAUSE TestAccAWSAPIGatewayV2Authorizer_basic
=== RUN   TestAccAWSAPIGatewayV2Authorizer_disappears
=== PAUSE TestAccAWSAPIGatewayV2Authorizer_disappears
=== RUN   TestAccAWSAPIGatewayV2Authorizer_Credentials
=== PAUSE TestAccAWSAPIGatewayV2Authorizer_Credentials
=== RUN   TestAccAWSAPIGatewayV2Authorizer_JWT
=== PAUSE TestAccAWSAPIGatewayV2Authorizer_JWT
=== RUN   TestAccAWSAPIGatewayV2Authorizer_HttpApiLambdaRequestAuthorizer
=== PAUSE TestAccAWSAPIGatewayV2Authorizer_HttpApiLambdaRequestAuthorizer
=== CONT  TestAccAWSAPIGatewayV2Authorizer_basic
=== CONT  TestAccAWSAPIGatewayV2Authorizer_JWT
=== CONT  TestAccAWSAPIGatewayV2Authorizer_HttpApiLambdaRequestAuthorizer
=== CONT  TestAccAWSAPIGatewayV2Authorizer_Credentials
=== CONT  TestAccAWSAPIGatewayV2Authorizer_disappears
    resource_aws_apigatewayv2_authorizer_test.go:58: [INFO] Got non-empty plan, as expected
--- PASS: TestAccAWSAPIGatewayV2Authorizer_disappears (57.32s)
--- PASS: TestAccAWSAPIGatewayV2Authorizer_basic (65.45s)
--- PASS: TestAccAWSAPIGatewayV2Authorizer_Credentials (86.23s)
--- PASS: TestAccAWSAPIGatewayV2Authorizer_JWT (91.33s)
--- PASS: TestAccAWSAPIGatewayV2Authorizer_HttpApiLambdaRequestAuthorizer (92.73s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	92.780s
```